sources/packages/backend/libs/utilities/src/date-utils.ts

```diff
@@ -246,3 +246,48 @@ export function hasSomePeriodOverlap(periods: Period[]): boolean {
   }
   return false;
 }
+
+/**
+ * Validates if the date provided is within given days
+ * before now.
+ * e.g. If the current date is 10-JAN-2023 and
+ * given number of days is 4, it validates if the given date
+ * is on or after 06-JAN-2023.
+ * @param givenDate date to be validated.
+ * @param numberOfDays number of days before today.
+ * @returns result if the given date is after given days before now.
+ */
+export const isAfterGivenDaysBeforeNow = (
+  givenDate: string | Date,
+  numberOfDaysBeforeToday: number,
+): boolean => {
+  return dayjs(givenDate).isAfter(
+    dayjs(dayjs(new Date()).format(DATE_ONLY_ISO_FORMAT)).subtract(
+      numberOfDaysBeforeToday + 1,
+      "days",
+    ),
+  );
+};
+
+/**
+ * Validates if the date provided is before
+ * given number of days from now.
+ * e.g. If the current date is 10-JAN-2023 and
+ * given number of days is 10, it validates if the given date
+ * is on or before 20-JAN-2023.
+ * @param givenDate date to be validated.
+ * @param numberOfDaysAfterToday number of days after today.
+ * @returns result if the given date is before given days from now.
+ */
+export const isBeforeGivenDaysFromNow = (
+  givenDate: string | Date,
+  numberOfDaysAfterToday: number,
+): boolean => {
+  return dayjs
+    .utc(givenDate)
+    .isBefore(
+      dayjs
```

Comment:
I would suggest creating this as a helper method in this data-utils or maybe reuse the method `getISODateOnlyString`.
```ts
dayjs .utc(dayjs.utc(new Date()).format("YYYY-MM-DD"))
```